### PR TITLE
Cache for cities and clubs

### PR DESCRIPTION
**Pull request by veronika-kolosova**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by Oleh-Bodnar — APPROVED** [review]

**Review by christykobyletska — APPROVED** [review]

**Review by sharix4 — APPROVED** [review]

**Review by danylo-melnychuk — APPROVED** [review]

File: EPlast/EPlast.Tests/Controllers/CityCacheControllerTests.cs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using EPlast.Resources;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Controllers;
+using Microsoft.AspNetCore.Routing;
+using EPlast.WebApi.Models.City;
+using System.Linq;
+using EPlast.DataAccess.Entities;
+using EPlast.BLL.DTO.City;
```
Comment (voroniak) [inline EPlast/EPlast.Tests/Controllers/CityCacheControllerTests.cs:18]: Sort usings

File: EPlast/EPlast.Tests/Controllers/ClubCacheControllerTests.cs [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+using Microsoft.AspNetCore.Http;
+using EPlast.Resources;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Controllers;
+using Microsoft.AspNetCore.Routing;
+using EPlast.WebApi.Models.Club;
+using System.Linq;
+using EPlast.DataAccess.Entities;
+using EPlast.BLL.DTO.Club;
+using EPlast.BLL.Interfaces.Club;
```
Comment (voroniak) [inline EPlast/EPlast.Tests/Controllers/ClubCacheControllerTests.cs:18]: sort usings

File: EPlast/EPlast.WebApi/Controllers/ClubCacheController.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+namespace EPlast.WebApi.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class ClubCacheController : Controller
+    {
+        private readonly IDistributedCache _cache;
+        private readonly IClubService _clubService;
+
+        public ClubCacheController(IClubService clubService,IDistributedCache cache)
```
Comment (voroniak) [inline EPlast/EPlast.WebApi/Controllers/ClubCacheController.cs:23]: format all files. use ctrl + k + d and ctrl + r + g

File: EPlast/EPlast.WebApi/Models/City/CitiesViewModel.cs [hunk 4]
```diff
@@ -6,11 +6,20 @@ namespace EPlast.WebApi.Models.City
...
-            CanCreate = isAdmin;
+            if (cityName == null)
+            {
+                Cities = cities.Skip((page - 1) * pageSize).Take(pageSize);
+                Total = cities.Count();
+                CanCreate = isAdmin;
+            }
+            else
+            {
+                Cities = from city in cities where city.Name.ToLower().Contains(cityName.ToLower()) select city;
```
Comment (voroniak) [inline EPlast/EPlast.WebApi/Models/City/CitiesViewModel.cs:19]: Why don't you use LINQ Extension Methods?

File: EPlast/EPlast.WebApi/Models/Club/ClubsViewModel.cs [hunk 5]
```diff
@@ -9,11 +9,20 @@ namespace EPlast.WebApi.Models.Club
...
-            CanCreate = isAdmin;
+            if (clubName == null)
+            {
+                Clubs = сlubs.Skip((page - 1) * pageSize).Take(pageSize);
+                Total = сlubs.Count();
+                CanCreate = isAdmin;
+            }
+            else
+            {
+                Clubs = from club in сlubs where club.Name.ToLower().Contains(clubName.ToLower()) select club;
```
Comment (voroniak) [inline EPlast/EPlast.WebApi/Models/Club/ClubsViewModel.cs:22]: Why don't you use LINQ Extension Methods?

**Review by voroniak — CHANGES_REQUESTED** [review]

**Comment by danylo-melnychuk** [comment]
@veronika-kolosova Pls, fix your conflicts and merge this pull request.

**Pull request closed without merging**